src/contexts/alert_handler.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { createContext, useContext, useState } from "react";
+
+const AlertHandlerContext = createContext();
+
+export const AlertHandlerProvider = ({ children }) => {
+  const [alertState, setAlertState] = useState({
+    visible: false,
+    errorText: "",
+  });
+
+  const setAlert = (errorText) => {
+    setAlertState({ ...alertState, visible: true, errorText: errorText });
+  };
+
+  const closeAlert = () => {
+    setAlertState({
+      ...alertState,
+      visible: false,
+      errorText: alertState.erroText,
+    });
+  };
+
+  const contextValue = {
+    ...alertState,
+    setAlert: setAlert,
+    closeAlert: closeAlert,
```

Comment:
```suggestion
    setAlert,
    closeAlert,
```
プロパティ名と値の変数名が同じ場合はこのように書くこともできます！